samcli/lib/utils/file_observer.py

```diff
@@ -0,0 +1,163 @@
+"""
+Wraps watchdog to observe file system for any change.
+"""
+import logging
+import threading
+
+from pathlib import Path
+from watchdog.observers import Observer
+from watchdog.events import PatternMatchingEventHandler
+
+from samcli.lib.utils.hash import dir_checksum, file_checksum
+
+LOG = logging.getLogger(__name__)
+
+
+class FileObserverException(Exception):
+    """
+    Exception raised when unable to observe the input path.
+    """
+
+
+class FileObserver:
```

Comment:
Thank you for abstracting this out now. This could come in handy if we ever want an auto/watch build command to be enabled! 🎉